### ISSUE TITLE
Certain 4k videos not working

### ISSUE THREAD

**Issue by Agentdouble046** [namidaco/namida#400]
I edit videos but certain 4k videos are not working it will say no_exceeds_capabilities this is the format of one video that doesn't work matroska,webm • stereo ch • 7542 kbps • 48.0 khz
mp4 and here's the resolution 
Resolution: 3840x2160
Bitrate: 7365 kb/s

**Comment by MSOB7YY** [comment]
do not expect exoplayer to play eveything, do u have a small sample? i will try it with ffmpeg as a default backend player

**Comment by Maxitendo1** [comment]
Some videos of mine are also not working, here's a sample:

https://drive.proton.me/urls/S34XJX6PYM#JjIUwh3GmzD1

**Comment by MSOB7YY** [comment]
@Maxitendo1 link doesnt open for me :(

**Comment by Maxitendo1** [comment]
@MSOB7YY Hopefully this one is working:

https://mega.nz/folder/CUlmmagK#12qfwd082t8lLKTDolWXlg

**Comment by MSOB7YY** [comment]
@Maxitendo1 yesss thanks will test soon

**Comment by MSOB7YY** [comment]
lmaoo it works on mine
![Screenshot_20250103-162521_Namida](https://github.com/user-attachments/assets/3b655d95-063a-45fd-abaf-0252e84f7ca0)


although i know that some videos wont play without ffmpeg set as primary decoder, i will try to see what do to

**Comment by MSOB7YY** [comment]
@Maxitendo1 @Agentdouble046 hey maybe check 5.0.5? i updated the underlying player and ffmpeg versions https://github.com/namidaco/namida-snapshots/releases

**Comment by Maxitendo1** [comment]
@MSOB7YY Unfortunately still not working. Here's the error output. I kinda got it working for one video but it's lagging very much if I play it.

![Image](https://github.com/user-attachments/assets/2c783d6b-12ef-4890-af1b-d81b3aa0d29e)

**Comment by Maxitendo1** [comment]
https://github.com/user-attachments/assets/a52f0731-1344-4753-9725-1fb353b01ac8

**Comment by Agentdouble046** [comment]
Im sorry but mines still not working here's a sample sorry it took so long 

https://www.mediafire.com/file/vj9ekhbtpeo3afq/MXClip_Geezy_Loc_-_Ghost_Of_Nyneteen_ft_A-killz_%252C_Doucemane_%252C%2526_Dre_Barrs_4K_HIGH_FR60-mc_2025_03_22_10_29_27-mc.mp4/file

**Comment by MSOB7YY** [comment]
@Maxitendo1 it still lags in fullscreen? if yes then probably this the best we can do

@Agentdouble046 even mx player couldn't play the video, it seems corrupted

**Comment by Agentdouble046** [comment]
Yeah my bad the file got corrupted before i uploaded here's the new link 

https://www.mediafire.com/file/f254tmxdyqgsr3k/Geezy_Loc_-_Ghost_Of_Nyneteen_ft_A-killz_%252C_Doucemane_%252C%2526_Dre_Barrs_4K_HIGH_FR60-mc/file


It shouldn't be corrupted now i checked

**Comment by Maxitendo1** [comment]
@MSOB7YY No, it still lags in general and this is the only one that works. You can't even play the 5 other ones that I have downloaded.

@Agentdouble046 I downloaded your video and I can confirm that I also can't play it.

**Comment by Agentdouble046** [comment]
@MSOB7YY i just downloaded it again it plays on mx player the link is working

**Comment by MSOB7YY** [comment]
@Agentdouble046 
i can play the video normally and without any lag
![Image](https://github.com/user-attachments/assets/74c0bab4-dc9b-409d-b309-ed737678d812)

**Comment by MSOB7YY** [comment]
> [@MSOB7YY](https://github.com/MSOB7YY) No, it still lags in general and this is the only one that works. You can't even play the 5 other ones that I have downloaded.
>

in that case i honestly dont know what else to do

soon i might have to use a different package for audio/video playback for windows support, i might allow a switch to use it for android

**Comment by Agentdouble046** [comment]
@MSOB7YY how im still getting the exceeds capabilities error btw im on the beta release

**Comment by Maxitendo1** [comment]
@MSOB7YY It's weird that every video is working on your Namida app, but not on ours.

**Comment by Maxitendo1** [comment]
Well...

https://github.com/google/ExoPlayer/issues/11216

**Comment by MSOB7YY** [comment]
@Maxitendo1 @Agentdouble046 not really weird, the reason is that ur device probably doesn't support that video format, normally it should fallback to ffmpeg extension (software decoder) but idk what happens with yall

let's wait till video playback for windows is implemented

**Comment by MSOB7YY** [comment]
> Well...
> 
> [google/ExoPlayer#11216](https://github.com/google/ExoPlayer/issues/11216)

the issue u mentioned is unrelated tho, these are some related ones https://github.com/androidx/media/issues/966 https://github.com/androidx/media/issues/372 https://github.com/androidx/media/issues/1311

**Comment by Maxitendo1** [comment]
Okay, thank you for the corrections, @MSOB7YY!

**Comment by Maxitendo1** [comment]
> [@MSOB7YY](https://github.com/MSOB7YY) No, it still lags in general and this is the only one that works. You can't even play the 5 other ones that I have downloaded.
> 
> [@Agentdouble046](https://github.com/Agentdouble046) I downloaded your video and I can confirm that I also can't play it.

For the new beta update 5.1.2 every 4K videos aren't putting out an error output anymore, which means that all of them are playing, but are still lagging and delayed.

**Comment by MSOB7YY** [comment]
@Maxitendo1 thanks for testing, it wasn't falling back to ffmpeg decoder because we had to explicitly enable "falling back" (https://github.com/MSOB7YY/just_audio/commit/d5c5137ba0ec38d55fa3ea7121ed0ef739e16b9c)

now the delay is mostly like that and can't be fixed, when u play in mx player, does it fallback to sw decoder? and how's the lag there?

**Comment by Maxitendo1** [comment]
@MSOB7YY MX Player works very well with falling back to sw coder, hw coder is not working at all or not even playing.

**Comment by MSOB7YY** [comment]
somehow for me it works very smooth on namida while lags a bit on mx player lmaoo (tested both videos)
not sure what's going wrong there

**Comment by MSOB7YY** [comment]
since the original issue is fixed, im going to close now. about possible lag with 4k+, if i got to know a solution about it i will fix it for sure, but currently thats how it is, sry for that
